platform/cxxsupport/mstd_atomic

```diff
@@ -23,7 +23,6 @@
 #include <mstd_memory>
 #include <mstd_type_traits>
 #include <mstd_utility>
-#include "platform/mbed_assert.h"
```

Comment:
I think the include should be kept.

Reply:
Why? Just to avoid compilation glitches from people who've been accidentally picking up `MBED_ASSERT` via an `<mstd_atomic>` include?

I agree this is a fairly common minor compatibility break - header implementations stopping including undocumented things - but don't recall it being a general rule not to do it.

In practice this is one of the less likely places to cause issues - there aren't many `<mstd_atomic>` users around, cos it's relatively new. You'd be far more likely to get headaches from losing something in `DigitalOut` or `SerialBase`, and I've not noticed reluctance to tidy their includes.

Although having said that, I don't see any other tidies in this PR. Not sure if that's because they couldn't be done or they just haven't been.